core/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -32,23 +32,17 @@ private[spark] trait Logging {
 
   // Make the log field transient so that objects with Logging can
   // be serialized and used on another machine
-  @transient private var log_ : Logger = null
+  @transient lazy val log: Logger = {
```

Comment:
I think `lazy` will add a `volatile` flag to `log` and prevents some JIT optimizations?